tests/integration/testdata/seccomp_syscall_test1.c

```diff
@@ -57,10 +57,10 @@ int main(void)
 	syscall_assert(raw(process_vm_writev, 0, NULL, 0, NULL, 0, ~0), -EPERM);
 
 	// Multiple arguments with AND rules.
-	syscall_assert(raw(kcmp, 0, 1337, 0, 0, 0), -ESRCH);
-	syscall_assert(raw(kcmp, 0, 0, 0, 0, 0), -EPERM);
-	syscall_assert(raw(kcmp, 500, 1337, 0, 0, 0), -EPERM);
-	syscall_assert(raw(kcmp, 500, 500, 0, 0, 0), -EPERM);
+	syscall_assert(raw(ftruncate, 123456789, 1337), -EBADF);
+	syscall_assert(raw(ftruncate, 0, 0), -EPERM);
+	syscall_assert(raw(ftruncate, 500, 1337), -EPERM);
+	syscall_assert(raw(ftruncate, 500, 500), -EPERM);
```

Comment:
I think the original idea here was to check that AND rules work. Meaning, (1) if arg0 is 0 and arg1 is 1337, the syscall is allowed (and we get a different error), while (2) if either arg0 is not 0 or arg1 is not 1337, we should get EPERM, and (3) if both arg0 and arg1 are not as expected, we also get EPERM.

This change removed the check for (2) for the case when arg0 is as expected and arg1 is not.

In other words, the second check for ftruncate should look like this:
```C
syscall_assert(raw(ftruncate, 123456789, 0), -EPERM);
```

(cc @cyphar)

Reply:
Fixed